Aws/S3/Commands/CopyObject.hs

```diff
@@ -55,12 +56,12 @@ instance SignQuery CopyObject where
     signQuery CopyObject {..} = s3SignQuery S3Query {
                                  s3QMethod = Put
                                , s3QBucket = Just $ T.encodeUtf8 coBucket
-                               , s3QObject = Just $ T.encodeUtf8 coObjectName
+                               , s3QObject = Just . HTTP.urlEncode False $ T.encodeUtf8 coObjectName
```

Comment:
Can you do the HTTP.urlEncode in s3SignQuery instead? Then it should not be necessary to do it in 5 different places.